cloud/scope/client.go

```diff
@@ -3,10 +3,27 @@ package scope
 import (
 	"context"
 
+	"github.com/linode/linodego"
 	"sigs.k8s.io/cluster-api/util/patch"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// LinodeObjectStorageClient defines functions suitable for provisioning object storage buckets and keys.
+type LinodeObjectStorageClient interface {
+	GetObjectStorageBucket(ctx context.Context, cluster, label string) (*linodego.ObjectStorageBucket, error)
+	CreateObjectStorageBucket(ctx context.Context, opts linodego.ObjectStorageBucketCreateOptions) (*linodego.ObjectStorageBucket, error)
+	CreateObjectStorageKey(ctx context.Context, opts linodego.ObjectStorageKeyCreateOptions) (*linodego.ObjectStorageKey, error)
+	DeleteObjectStorageKey(ctx context.Context, keyID int) error
+}
+
+// LinodeObjectStorageClientBuilder is a function that returns a LinodeObjectStorageClient.
+type LinodeObjectStorageClientBuilder func(apiKey string) (LinodeObjectStorageClient, error)
+
+// CreateLinodeObjectStorageClient is the main implementation of LinodeObjectStorageClientBuilder.
+func CreateLinodeObjectStorageClient(apiKey string) (LinodeObjectStorageClient, error) {
+	return CreateLinodeClient(apiKey)
+}
```

Comment:
Question: What is this func being used for?

Reply:
I added this since we don't want to use an anonymous function [here](https://github.com/linode/cluster-api-provider-linode/blob/befb7302c346c65168ba044e4f10e064e7ce583e/cmd/main.go#L150) (to pass golangci-lint rules).